pkg/server/projectconfig/service.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package projectconfig
+
+import (
+	"errors"
+
+	util "github.com/daytonaio/daytona/internal/util"
+	"github.com/daytonaio/daytona/pkg/server/projectconfig/dto"
+	"github.com/daytonaio/daytona/pkg/workspace/project/config"
+)
+
+type IProjectConfigService interface {
+	Delete(projectConfigName string) error
+	Find(projectConfigName string) (*config.ProjectConfig, error)
+	FindDefault(url string) (*config.ProjectConfig, error)
+	List() ([]*config.ProjectConfig, error)
+	FilterByGitUrl(url string) ([]*config.ProjectConfig, error)
```

Comment:
I see that you added `url string` as an argument. I believe that a `Filter` struct would be more appropriate since it's more extensible that way.

Reply:
I haven't seen this elsewhere in the project, the upcoming build poller uses this only on db level.
Since additional filters likely won't be needed soon I'd postpone this for then